compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -5346,11 +5346,18 @@ private BLangLiteral createSimpleLiteral(Node literal, SyntaxKind sign, boolean
                 //TODO: Check effect of mapping negative(-) numbers as unary-expr
                 kind = NodeKind.DECIMAL_FLOATING_POINT_LITERAL;
                 typeTag = NumericLiteralSupport.isDecimalDiscriminated(textValue) ? TypeTags.DECIMAL : TypeTags.FLOAT;
+                String val = textValue;
+                
+                // Remove exponent indicator if it is not followed by a digit
+                if (val.matches(".*[eE][+-]?[fd]?")) {
+                    val = val.replaceAll("[eE][+-]?", "");
+                }
```

Comment:
Consider the following two examples
Example 1
```ballerina
public function main() {
    10f a = 10Ef ; 
}
```
Gives error as
```
ERROR [test3.bal:(23:13,23:17)] missing digit after exponent indicator
```
Example 2
```ballerina
public function main() {
    10f a = 100Ef ; 
}
```
Gives error as
```
ERROR [test3.bal:(23:13,23:18)] incompatible types: expected '10ff', found 'float'
ERROR [test3.bal:(23:13,23:18)] missing digit after exponent indicator
```
In the first example, identify the `10Ef ` as `10f`(due to that didn't we get incompatible types error). It's not correct. right?
In the second example,  identify the `100Ef ` as `100f`(due to that we get an incompatible types error).
IMO we cannot set the type for incorrect values. (In the first example we set the type of `10Ef ` as `10f `)

Reply:
We can move this logic to Types.validateFloatLiteral() and return `false` there for invalid literals which will fix the above issue. Should we go for it?

Reply:
I feel giving the syntax error is enough in these scenarios. One has to fix the syntax error first.

We can simply do the following at L5370, to avoid any crashes in the compiler phases. WDYT?

```java
            if (literal.hasDiagnostics()) {
                value = "0";
                originalValue = "0";
            }
```

Reply:
Another option would be to use a try-catch for `Double.parseDouble()` and if an exception is thrown we simply return false. But, the problem there is we can't guarantee the syntax error is there. Ideally, there should be a syntax error if `Double.parseDouble()` is failing though.

Reply:
setting value to `"0"` will still give the incompatible type error right?

Reply:
We don't need try-catch right? we have `Types.validateFloatLiteral()` method so we can include those checks to it and return false there

Reply:
```bal
public function main() {
   "A" a = "\u{0041";
}
```
The above sample also gives `incompatible types: expected 'A', found 'string'` error on an invalid string literal

Reply:
Fixed in [932a51e](https://github.com/ballerina-platform/ballerina-lang/pull/35646/commits/932a51ecb0f19f3946d3394d71c2ed14ad5dcb73)

Reply:
Created a separate issue for string case #35794